controlplane/internal/controller/clusterdeployment_controller.go

```diff
@@ -88,44 +89,94 @@ func (r *ClusterDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Re
+		return ctrl.Result{}, err
+	}
+
+	if acp.Spec.Config.ImageRegistryRef != nil {
+		if err := r.createImageRegistry(ctx, acp.Spec.Config.ImageRegistryRef.Name, acp.Namespace); err != nil {
+			log.Error(err, "failed to create image registry config manifest")
+			return ctrl.Result{}, err
+		}
+	}
+
+	if err := r.ensureAgentClusterInstall(ctx, clusterDeployment, acp); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, r.updateClusterDeploymentRef(ctx, clusterDeployment)
 }
 
 func (r *ClusterDeploymentReconciler) ensureAgentClusterInstall(
 	ctx context.Context,
 	clusterDeployment *hivev1.ClusterDeployment,
 	oacp controlplanev1alpha2.OpenshiftAssistedControlPlane,
-) (ctrl.Result, error) {
+) error {
 	log := ctrl.LoggerFrom(ctx)
 
 	cluster, err := capiutil.GetOwnerCluster(ctx, r.Client, oacp.ObjectMeta)
 	if err != nil {
 		log.Error(err, "failed to retrieve owner Cluster from the API Server")
-		return ctrl.Result{}, err
+		return err
 	}
 
-	arch, err := getArchitectureFromBootstrapConfigs(ctx, r.Client, &oacp)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-	imageSet := computeClusterImageSet(clusterDeployment.Name, getReleaseImage(oacp, arch))
-	err = util.CreateOrUpdate(ctx, r.Client, imageSet)
-	if err != nil {
-		log.Error(err, "failed creating ClusterImageSet")
-		return ctrl.Result{}, err
+	workerNodes := r.getWorkerNodesCount(ctx, cluster)
+	clusterNetwork, serviceNetwork := getClusterNetworks(cluster)
+	additionalManifests := getClusterAdditionalManifestRefs(oacp)
+
+	aci := &hiveext.AgentClusterInstall{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterDeployment.Name,
+			Namespace: clusterDeployment.Namespace,
+		},
 	}
+	mutate := func() error {
+		aci.ObjectMeta.Labels = util.ControlPlaneMachineLabelsForCluster(
+			&oacp,
+			clusterDeployment.Labels[clusterv1.ClusterNameLabel],
+		)
+		aci.ObjectMeta.Labels[hiveext.ClusterConsumerLabel] = openshiftAssistedControlPlaneKind
+		aci.ObjectMeta.OwnerReferences = []metav1.OwnerReference{
+			*metav1.NewControllerRef(&oacp, controlplanev1alpha2.GroupVersion.WithKind(openshiftAssistedControlPlaneKind)),
+		}
 
-	workerNodes := r.getWorkerNodesCount(ctx, cluster)
-	aci, err := r.computeAgentClusterInstall(ctx, clusterDeployment, oacp, imageSet, cluster, workerNodes)
-	if err != nil {
-		return ctrl.Result{}, err
+		aci.Spec.ClusterDeploymentRef = corev1.LocalObjectReference{Name: clusterDeployment.Name}
+		aci.Spec.PlatformType = hiveext.PlatformType(configv1.NonePlatformType)
+		aci.Spec.ProvisionRequirements = hiveext.ProvisionRequirements{
+			ControlPlaneAgents: int(oacp.Spec.Replicas),
+			WorkerAgents:       workerNodes,
+		}
+		aci.Spec.DiskEncryption = oacp.Spec.Config.DiskEncryption
+		aci.Spec.MastersSchedulable = oacp.Spec.Config.MastersSchedulable
+		aci.Spec.Proxy = oacp.Spec.Config.Proxy
+		aci.Spec.SSHPublicKey = oacp.Spec.Config.SSHAuthorizedKey
+		aci.Spec.ImageSetRef = &hivev1.ClusterImageSetReference{Name: clusterDeployment.Name}
```

Comment:
I think the ClusterImageSet name is based off of the OACP spec 

```suggestion
                 if oacp.Spec.Config.ImageRegistryRef != nil {
		    aci.Spec.ImageSetRef = &hivev1.ClusterImageSetReference{Name: acp.Spec.Config.ImageRegistryRef.Name}
		}
```

Reply:
hmmm maybe it should be but I'm pretty sure what I did here is the same as what we started with.

You can see it a bit more easily if you look at each commit, but I know there ended up being a lot of them :sweat_smile: 

In current master we pass `clusterDeployment.Name` into `computeClusterImageSet` which uses it directly as a parameter to name the image set.

Reply:
And to take it further `computeAgentClusterInstall` takes in the `imageSet` object which is used for its name.

Reply:
Oh 🤦‍♀️ I was looking at the wrong function, sorry! That's right it is the cluster deployment name, sorry ignore this 😓